File: Homework/Khang_Nguyen/reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 function recursion(head, prev) {
    if (head === null) return null
    if (head.next === null) {
        head.next = prev
        return head
    }
    
    const next = head.next
    head.next = prev
    return result = recursion(next, head)
}
var reverseList = function(head) {
    return recursion(head, null)
};

File: Homework/Van_Le/lesson1/ReverseLinkedList/ReverseLinkedList.cs
namespace LinkedList
{
    public class ReverseLinkedList
    {
        public ListNode ReverseList(ListNode head)
        {
            ListNode prev = null, current = head;
            while (current != null)
            {
                ListNode next = current.next;
                current.next = prev;
                prev = current;
                current = next;
            }
            head = prev;
            return head;
        }
    }
}
